tests/test_benchmark.py

```diff
@@ -229,6 +229,86 @@ def benchmark(
     )
 
 
+def benchmark_local(
+    label,
+    masses,
+    x0,
+    v0,
+    box,
+    bound_potentials,
+    ligand_idxs,
+    hmr=False,
+    verbose=True,
+    num_batches=100,
+    steps_per_batch=1000,
+):
+    """
+    TODO: configuration blob containing num_batches, steps_per_batch, and any other options
+    """
+
+    seed = 1234
+    dt = 1.5e-3
+    temperature = constants.DEFAULT_TEMP
+    friction = 1.0
+    seconds_per_day = 86400
+
+    harmonic_bond_potential = bound_potentials[0]
+    bond_list = get_bond_list(harmonic_bond_potential)
+    if hmr:
+        dt = 2.5e-3
+        masses = apply_hmr(masses, bond_list)
+    intg = LangevinIntegrator(temperature, dt, friction, np.array(masses), seed).impl()
+
+    bps = []
+
+    for potential in bound_potentials:
+        bps.append(potential.bound_impl(precision=np.float32))  # get the bound implementation
+
+    ctxt = custom_ops.Context(
+        x0,
+        v0,
+        box,
+        intg,
+        bps,
+    )
+
+    batch_times = []
+
+    ligand_idxs = ligand_idxs.astype(np.int32)
+
+    # run once before timer starts
+    ctxt.multiple_steps_local(steps_per_batch, ligand_idxs, burn_in=0)
+
+    start = time.time()
+
+    for batch in range(num_batches):
+
+        # time the current batch
+        batch_start = time.time()
+        _, _ = ctxt.multiple_steps_local(steps_per_batch, ligand_idxs, burn_in=0)
```

Comment:
clarification: should a different random seed be set each batch?

Reply:
Fixed in [7c8d6f1](https://github.com/proteneer/timemachine/pull/968/commits/7c8d6f161b6b3545df69eff2c85a3d78ce4bb982) and [0088560](https://github.com/proteneer/timemachine/pull/968/commits/00885600ca54445690a8abf5bf76e6f144531f33)